ts/main.ts

```diff
@@ -1,64 +1,170 @@
+}
 
+if (!$divEntries || !$divEntryForm) {
+  throw new Error('$divEntries or !$divEntryForm not exist');
+}
 
-$inputURL.addEventListener('input', (event: Event) => {
+if (!$inputURL || !$img || !$form) {
+  throw new Error('$inputURL or $img or $form are not exists');
+}
 
+$inputURL.addEventListener('input', (event: Event) => {
   const eventTarget = event.target as HTMLInputElement;
 
-
   const imgURL = eventTarget.value;
 
-
   $img.src = imgURL;
-
 });
 
-
 $form.addEventListener('submit', (event: Event) => {
-
   event.preventDefault();
-
-  const entry: Entry = {};
-
   const formElements = $form.elements;
 
   const formTitle = $form.elements[0] as HTMLInputElement;
   const formURL = $form.elements[1] as HTMLInputElement;
   const formTextArea = $form.elements[2] as HTMLTextAreaElement;
 
+  const entry: Entry = {
+    entryTitle: formTitle.value,
+    entryURL: formURL.value,
+    entryTextArea: formTextArea.value,
+    entryId: data.nextEntryId,
+  };
 
-  entry.entryTitle = formTitle.value;
-  entry.entryURL = formURL.value;
-  entry.entryTextArea = formTextArea.value;
-  entry.entryId = data.nextEntryId;
   data.nextEntryId = data.nextEntryId + 1;
-  // adding new post to te end which is wrong because new posts have to be at the top
-  // data.entries.push(entry);
-  // adding the new post data to the top, so we know its a new post
+
+  // every time we hit submit, a DOM tree is created or one entry li and appending it to ul a list
+  // of entries
+  // so its created quickly after submit which solves the issue of refreshing
+  // adding the new post data to the top or beginning, so we know its a new post
   data.entries.unshift(entry);
 
+  // adding only the newly created entry not the whole entries
+  // prepend adds the new li at the beginning not at the end which is similar to unshift()
+  // to maintain the same order in the DOM or html file like in the array entries
+  $ulList.prepend(renderEntry(entry));
 
-  writeEntry();
-  $img.src = "images/placeholder-image-square.jpg"
+  writeData();
+  $img.src = 'images/placeholder-image-square.jpg';
   $form.reset();
+  // when the form reset it immediately goes to entries
+  viewSwap('entries');
+  toggleNoEntries();
+});
+
+// from somewhere an entry will be passed to renderEntry function
+// returning an entry
+function renderEntry(entry: Entry): HTMLLIElement {
+  const $parentLI = document.createElement('li');
+  const $divRow = document.createElement('div');
+  $divRow.setAttribute('class', 'row');
+  // after being appended, it will be outputted because its added to the DOM
+  $parentLI.appendChild($divRow);
+  const $divColHalfFirst = document.createElement('div');
+  $divColHalfFirst.setAttribute('class', 'column-half');
+  $divRow.appendChild($divColHalfFirst);
+  const $img = document.createElement('img');
+  $img.setAttribute('class', 'entries-img');
+  $img.setAttribute('src', entry.entryURL);
+  $img.setAttribute('alt', 'mountain view');
```

Comment:
You don't want to give them all the same `alt` attribute, maybe use the `entry.title`